executor/app.py

```
@@ -97,21 +97,60 @@ async def lifespan(app: FastAPI):
+            auth_token = os.getenv("AUTH_TOKEN")
+            logger.info(
+                f"TASK_ID and AUTH_TOKEN found, fetching task info from API for task {task_id}"
+            )
+
+            from executor.services.api_client import fetch_task_execution_info
+
+            task_info = fetch_task_execution_info(task_id, auth_token)
+
+            if task_info:
+                # DEBUG: Save fetched task info to file for comparison
+                debug_file = f"/tmp/task_info_api_{task_id}.json"
```

Ruff (0.15.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/task_info_api_"

(S108)